src/engine/guidance/lane_processing.cpp

```diff
@@ -166,7 +166,7 @@ std::vector<RouteStep> anticipateLaneChange(std::vector<RouteStep> steps,
             // step as invalid, scheduled for later removal.
             if (collapsable(previous, current))
             {
-                previous = elongate(previous, current);
+                previous.ElongateBy(current);
```

Comment:
All functions above mutate a RouteStep in-place. For example here we also mutate in-place. What's the reason for returning *this by ref? So we can chain calls?